pkg/optimize/sg/sg.go

```diff
@@ -0,0 +1,215 @@
+/*
+Copyright 2023- IBM Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package sgoptimizer
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/np-guard/models/pkg/ds"
+	"github.com/np-guard/models/pkg/netp"
+	"github.com/np-guard/models/pkg/netset"
+
+	"github.com/np-guard/vpc-network-config-synthesis/pkg/ir"
+	"github.com/np-guard/vpc-network-config-synthesis/pkg/optimize"
+	"github.com/np-guard/vpc-network-config-synthesis/pkg/utils"
+)
+
+type (
+	sgOptimizer struct {
+		sgCollection *ir.SGCollection
+		sgName       ir.SGName
+		sgVPC        *string
```

Comment:
Consider making `sgVPC` a `string` rather than a pointer to a `string`, with the empty string marking no vpc.
In general, VPC names cannot be empty strings.